kart/spatial_tree.py

```diff
@@ -396,125 +387,261 @@ def _find_geometry_column(fields):
     """
-    # Note: the x-values have a valid range of 720 degrees at the equator. We could get an extra bit of
-    # precision if we gave them a valid range of only 360 degrees and stored max-x as an offset from min-x, but
-    # it makes the code more complicated.
-    result = _encode_value(envelope[0], -180, 540, math.floor)
-    result <<= BITS_PER_VALUE
-    result |= _encode_value(envelope[1], -180, 540, math.ceil)
-    result <<= BITS_PER_VALUE
-    result |= _encode_value(envelope[2], -90, 90, math.floor)
-    result <<= BITS_PER_VALUE
-    result |= _encode_value(envelope[3], -90, 90, math.ceil)
-    assert 0 <= result <= MAX_ENCODED_ENVELOPE
-    return result.to_bytes(BYTES_PER_ENVELOPE, BYTE_ORDER)
-
-
-def _encode_value(value, min_value, max_value, round_fn):
-    normalised = (value - min_value) / (max_value - min_value)
-    encoded = round_fn(normalised * MAX_ENCODED_VALUE)
-    assert 0 <= encoded <= MAX_ENCODED_VALUE
-    return encoded
-
-
-def decode_envelope(encoded):
-    """Inverse of encode_envelope."""
-    encoded = int.from_bytes(encoded, BYTE_ORDER)
-    assert 0 <= encoded <= MAX_ENCODED_ENVELOPE
-    max_y = _decode_value(encoded & MAX_ENCODED_VALUE, -90, 90)
-    encoded >>= BITS_PER_VALUE
-    min_y = _decode_value(encoded & MAX_ENCODED_VALUE, -90, 90)
-    encoded >>= BITS_PER_VALUE
-    max_x = _decode_value(encoded & MAX_ENCODED_VALUE, -180, 540)
-    encoded >>= BITS_PER_VALUE
-    min_x = _decode_value(encoded & MAX_ENCODED_VALUE, -180, 540)
-    return min_x, max_x, min_y, max_y
-
-
-def _decode_value(encoded, min_value, max_value):
-    assert 0 <= encoded <= MAX_ENCODED_VALUE
-    normalised = encoded / MAX_ENCODED_VALUE
-    return normalised * (max_value - min_value) + min_value
+    return envelope[0], envelope[2], envelope[1], envelope[3]
 
 
-INF = float("inf")
+def transform_minmax_envelope(minmax_envelope, transform):
+    """
+    Given an envelope in (min-x, min-y, max-x, max-y) format in any CRS, transforms it to EPSG:4326 using the given
+    transform, then returns an axis-aligned envelope in EPSG:4326 in (w, s, e, n) order that bounds the original
+    (but which may have a slightly larger area due to the axis-aligned edges not lining up with the original).
+    The returned envelope has w <= e unless it crosses the antimeridian, in which case e < w.
+    """
 
+    # FIXME: This function is not 100% correct - it should segmentize each edge of the envelope and transform
+    # all the segments, to account for the fact that these edges should sometimes curve when reprojected.
+    # But right now we just reproject the corners and assume straight lines between them - in rare cases the result
+    # won't cover the entire geometry.
+
+    ring = anticlockwise_ring_from_minmax_envelope(minmax_envelope)
+    ring.Transform(transform)
+    ring_points = [ring.GetPoint_2D(i) for i in range(5)]
+
+    if _is_clockwise(ring_points):
```

Comment:
For some reason it didn't error - maybe since I never do anything to the ring except transform it (eg I never add it to a ogr.wkbPolygon, or anything like that). But anyway I fixed it so that points are transformed in a different code path.